applications/nrf5340_audio/src/bluetooth/bt_management/scanning/bt_mgmt_scan_for_broadcast.c

```diff
@@ -42,6 +42,33 @@ struct broadcast_source {
 	uint32_t broadcast_id;
 };
 
+static void scan_restart_worker(struct k_work *work)
+{
+	int ret;
+
+	/* Delete pending PA sync before restarting scan */
+	ret = bt_mgmt_pa_sync_delete(pa_sync);
+	if (ret) {
+		LOG_WRN("Failed to delete pending PA sync: %d", ret);
+	}
+
+	ret = bt_mgmt_scan_start(0, 0, BT_MGMT_SCAN_TYPE_BROADCAST, NULL);
+	if (ret) {
+		LOG_WRN("Failed to restart scanning for braodcast: %d", ret);
+	}
+}
+
+K_WORK_DEFINE(scan_restart_work, scan_restart_worker);
```

Comment:
Probably we can move all the K_WORK_DEFINE to the top of the file.

Reply:
They have to be under the function that they call, to either we have to add a prototype at the top of the file or we keep the K_DEFINEs were they are. I had a look in the rest of our code, and K_WORKs and K_TIMERs are all in the midst of the other code (except one K_TIMER_DEFINE in bt_mgmt_ctlr_cfg.c were we have a prototype)